tests/testthat/test-formatting.R

```diff
@@ -112,3 +112,137 @@ test_that("format_na_str functionality works in get_formatted_cells (ie printing
                      "Ridiculous")
 })
+
+test_that("Test matrix obtained from get_formatted_cells as is into ASCII format", {
+
+  # Table builder with rtables
+  tbl <- basic_table() %>%
+    split_cols_by("Species") %>%
+    analyze(c("Sepal.Length", "Petal.Width"),
+      afun = mean,
+      format = NULL
+    ) %>% # NULL will be picked by format_rcell
+    build_table(iris)
+
+  # Get the ASCII table
+  result <- get_formatted_cells(tbl) # Main function
+
+  # Expected data-set is built with dplyr
+  tibl <- iris %>%
+    group_by(Species) %>%
+    summarise(
+      m_sep = mean(Sepal.Length),
+      m_pet = mean(Petal.Width)
+    ) %>%
+    select(-Species)
+
+  # Get it into an ASCII-like format
+  expected <- tibl %>%
+    mutate_all(as.character) %>%
+    mutate(a = "", b = "") %>%
+    select(a, m_sep, b, m_pet) %>%
+    t() %>%
+    as.matrix()
+
+  dimnames(expected) <- NULL # fixing attributes
+
+  # Check that preserve the format "as is"
+  expect_identical(result, expected)
+})
+
+test_that("Test format matrix obtained from get_formatted_cells with only NULL format", {
+
+  # Table builder with rtables
+  tbl <- basic_table() %>%
+    split_cols_by("Species") %>%
+    analyze(c("Sepal.Length", "Petal.Width"),
+      afun = mean,
+      format = NULL
+    ) %>%
+    build_table(iris)
+
+  # Get the ASCII table
+  result <- get_formatted_cells(tbl, shell = TRUE) # Main function
+
+  # Creating moking data
+  col_std <- c("-", "xx", "-", "xx")
+  expected <- cbind(col_std, col_std, col_std)
+
+  dimnames(expected) <- NULL # fixing attributes
+
+  # Check that preserve the format "as is"
+  expect_identical(result, expected)
+})
+
+test_that("Test matrix obtained from get_formatted_cells with only specific format", {
+
+  # Table builder with rtables
+  tbl <- basic_table() %>%
+    split_cols_by("Species") %>%
+    analyze(c("Sepal.Length", "Petal.Width"),
+      afun = mean,
+      format = "xx.xxx"
+    ) %>%
+    build_table(iris)
+
+  # Get the ASCII table
+  result <- get_formatted_cells(tbl, shell = TRUE) # Main function
+
+  # Creating moking data
+  col_std <- c("-", "xx.xxx", "-", "xx.xxx")
+  expected <- cbind(col_std, col_std, col_std)
+
+  dimnames(expected) <- NULL # fixing attributes
+
+  # Check that preserve the specific
+  expect_identical(result, expected)
+})
+
+test_that("Test matrix obtained from get_formatted_cells with only specific na values", {
+
+  # Introducing NAs
+  iris2 <- iris
+  iris2$Sepal.Length[sample(seq_len(nrow(iris2)), size = 10)] <- NA
+  iris2$Petal.Width[sample(seq_len(nrow(iris2)), size = 10)] <- NA
+
+  # Custom summary function to add specific formats for na values to be reflected
+  s_summary <- function(x) {
+    stopifnot(is.numeric(x))
+    list(
+      mean = mean(x)
+    )
+  }
+
+  afun <- make_afun(
+    fun = s_summary,
+    .formats = c(mean = "xx.xx"),
+    .labels = c(mean = "Mean")
+  )
+
+  afun_isridiculous <- make_afun(afun,
```

Comment:
Typically we don't use make_afun in tests unless we are _testing `make_afun` itself_, as it embodies a (very large) amount of additional complexity that is (usually) not relevant for what we're trying to test.

That said, we do also need to maintain the tests for `make_afun` so that coverage is good.

Reply:
I did use that to test that the na_str was propagated by the CellValues object instead of the previous way (through `get_formatted_cells`)